src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -32,19 +32,29 @@ public void TestEnvironmentProperty()
 
             IDictionary<string, string> environment = psi.Environment;
 
-            Assert.NotEqual(environment.Count, 0);
+            Assert.NotEqual(0, environment.Count);
 
-            int CountItems = environment.Count;
+            int countItems = environment.Count;
 
             environment.Add("NewKey", "NewValue");
             environment.Add("NewKey2", "NewValue2");
 
-            Assert.Equal(CountItems + 2, environment.Count);
+            Assert.Equal(countItems + 2, environment.Count);
             environment.Remove("NewKey");
-            Assert.Equal(CountItems + 1, environment.Count);
+            Assert.Equal(countItems + 1, environment.Count);
 
-            //Exception not thrown with invalid key
-            Assert.Throws<ArgumentException>(() => { environment.Add("NewKey2", "NewValue2"); });
+            // The full .NET framework represents Environment with StringDictionary. On .NET Core we use
+            // use Dictionary<string, string>, because we need to have case sensitive behaviour on Unix.
```

Comment:
The default behaviour of StringDictionary is case insensitive. The full .NET framework uses StringDictionary internally to hold Environment. 
There was no good way to make StringDictionary case insensitive without accessing internal state of StringDictionary, which we can't do in the .NET Core packaged world.

I assumed this is why we went with Dictionary. Maybe I'm wrong!